pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -98,7 +98,7 @@ in rec {
     let
       onPath = f: a: if typeOf a == "path" then f a else a;
       str_patterns = map (onPath readFile) (lib.toList file_str_patterns);
-    in concatStringsSep "\n" str_patterns;
+    in ".git\n" + (concatStringsSep "\n" str_patterns);
```

Comment:
I think it indeed makes some sense to add `.git` to the default filters for `gitignoreSource`.

This doesn't seem to be the right place to put this default value to me, though:

The 2 documented entry points for nix-gitignore are `gitignoreSource` and `gitignoreSourcePure`. They both call `gitignoreFilterSourcePure`, but `gitignoreSource` adds `.gitignore` to the parameters (via `withGitignoreFile`), while `gitignoreSourcePure` does not.

To be consistent I think it the additional default `.git` ignore should only be added to `gitignoreSource`, and `gitignoreSourcePure` should remain 'pure'. So I think `.git` should be added to `withGitignoreFile` (perhaps renamed to `withDefaults` or something?) instead of `gitignoreCompileIgnore`

Reply:
My rationale was: "`gitignoreSource` works like git and `gitignoreSourcePure` like git in a repo where there is no `.gitignore`". What do you think?

Reply:
That's a bit strange when there actually is a `.gitignore` present in that directory, though, right?

To me the interpretation '`gitignoreSource` has defaults to work like git and `gitignoreSourcePure` gives you complete control' makes more sense.

Reply:
Ok makes sense, I changed the PR to only affect the non pure variant.